src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs

```diff
@@ -138,13 +140,26 @@ internal override void ProcessResponse(HttpResponseMessage response)
                     WriteObject(obj);
                 }
             }
-            else if (ShouldSaveToOutFile)
+            
+            if (ShouldSaveToOutFile)
             {
                 string outFilePath = WebResponseHelper.GetOutFilePath(response, _qualifiedOutFile);
+                Stream saveStream;
 
-                WriteVerbose(string.Create(System.Globalization.CultureInfo.InvariantCulture, $"File Name: {Path.GetFileName(_qualifiedOutFile)}"));
+                // Saves the response stream to a file. If piped then the response stream needs to be rewinded.
+                if (responseStream != null)
+                {
+                    responseStream.Position = 0;
```

Comment:
I suggest to use the same code pattern we used for Invoke-WebRequest
https://github.com/PowerShell/PowerShell/blob/0f13f452e346211abee02e5dace5665c97a5af46/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/InvokeWebRequestCommand.CoreClr.cs#L61-L72

Reply:
Thank you for the suggestion! It took me one more look to realize how similar it can be. Changes made.